modules/s2i/bash/artifacts/usr/local/s2i/assemble

```diff
@@ -14,3 +14,52 @@ source "${JBOSS_CONTAINER_JAVA_S2I_MODULE}/maven-s2i-overrides"
 # invoke the build
 maven_s2i_build
 
+#TODO: Clean this up to call the scripts instead
+
+shopt -s globstar
+echo "Setting environment variables"
+jarfile="/deployments/*.jar"
+libdir="/deployments/lib"
+$JAVA_HOME/
+
+echo "Checking that jarfile and libdir exist"
+test -f "$jarfile"
+test -d "$libdir"
+
+# Create a temporary directory for a module path
+mkdir dependencies
+cp /deployments/lib/**/*.jar dependencies
+# Calculate dependencies
+$JAVA_HOME/bin/jdeps --module-path dependencies --ignore-missing-deps --multi-release 11 -R -s \
```

Comment:
The hard-coded 11 here, will that need to be 17 (or 21) etc? We can rely on `$JAVA_VERSION` being set correctly in the run context inside the container; perhaps use that